src/state/opamSwitchState.ml

```diff
@@ -1438,3 +1432,29 @@ let reverse_dependencies st ~build ~post =
            | Some nv -> OpamPackage.Set.add nv result
            | None -> OpamStd.Sys.exit_because `Internal_error)
          int_revdeps packages)
+
+
+(* invariant computation *)
+
+let invariant_root_packages st =
+  OpamPackage.Set.filter (OpamFormula.verifies st.switch_invariant) st.installed
+
+let compute_invariant_packages st =
+  let pkgs = invariant_root_packages st in
+  dependencies ~build:false ~post:false ~depopts:false ~installed:true
```

Comment:
Shouldn't this rather be `~build:true ~post:true` ? It seems at least the post dependencies could legitimately be consideded part of the base.

Reply:
true